### PR TITLE
[device/celestica]: Add thermalctld support on Haliburton platform APIs

### PR DESCRIPTION
**Pull request by mudsut4ke**
**- Why I did it**
- The thermalctld daemon on the Pmon docker requires support from the thermal manager API.

**- How I did it**
- Removed the old function for detecting a faulty fan.
- Removed the old function for detecting excess temperature.
- Implement thermal_manager APIs based on ThermalManagerBase
- Implement thermal_conditions APIs based on ThermalPolicyConditionBase
- Implement thermal_actions APIs based on ThermalPolicyActionBase
- Implement thermal_info APIs based on ThermalPolicyInfoBase
- Add thermal_policy.json

**- How to verify it**
- Check the fan speed during temperature changes.
- Examine events that will occur after the temperature has exceeded the threshold.
- Check for events that will occur after the fan is removed or the fan is not working properly.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [x] 201911
- [x] 202006

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-cel/debian/platform-modules-haliburton.init [hunk 1]
```diff
@@ -54,8 +54,8 @@ start)
         echo max6699 0x1a > /sys/bus/i2c/devices/i2c-11/new_device
 
         # Attach PSUs
-        echo dps200 0x5a > /sys/bus/i2c/devices/i2c-12/new_device
-        echo dps200 0x5b > /sys/bus/i2c/devices/i2c-13/new_device
+        # echo dps200 0x5a > /sys/bus/i2c/devices/i2c-12/new_device
+        # echo dps200 0x5b > /sys/bus/i2c/devices/i2c-13/new_device
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-cel/debian/platform-modules-haliburton.init:58]: Remove commented code or add a `TODO` comment if you plan to uncomment later

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/broadcom/sonic-platform-modules-cel/debian/platform-modules-haliburton.init (hunk 1 above)
Comment (mudsut4ke) [reply platform/broadcom/sonic-platform-modules-cel/debian/platform-modules-haliburton.init:61]: updated: https://github.com/Azure/sonic-buildimage/pull/6493/commits/0664a4fb56d29fa4fd8cb430a11a487822480abd

**Comment by mudsut4ke** [comment]
retest kvmtest-t0 please

**Comment by mudsut4ke** [comment]
@jleveque , Please help to re-review this PR

**Comment by mudsut4ke** [comment]
@jleveque , Can you merge this PR ?

**Comment by jleveque** [comment]
@mudsut4ke: Should this PR also be cherry-picked to the 202012 branch?

**Review by jleveque — APPROVED** [review]

**Comment by mudsut4ke** [comment]
> @mudsut4ke: Should this PR also be cherry-picked to the 202012 branch?
@jleveque , Yes, can you please cherry-picked to the 202012 branch

File: device/celestica/x86_64-cel_e1031-r0/sonic_platform/common.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+        Args:
+            config_name: A string containing the name of config file.
+
+        Returns:
+            A string containing the path to json file
+        """
+        return os.path.join(self.DEVICE_PATH, self.platform, self.CONFIG_DIR, config_name) if self.is_host() else os.path.join(self.PMON_PLATFORM_PATH, self.CONFIG_DIR, config_name)
+
+    def get_output(self, index, config, default):
```
Comment (qiluo-msft) [inline device/celestica/x86_64-cel_e1031-r0/sonic_platform/common.py:195]: >get_output [](http://example.com/codeflow?start=8&length=10)

Seems this function is not used. Please remove.
@mudsut4ke Could you help review?

File: device/celestica/x86_64-cel_e1031-r0/sonic_platform/common.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            hex_ver = self.get_reg(path, addr)
+            output = self._hex_ver_decode(
+                hex_ver, config['num_of_bits'], config['num_of_points'])
+
+        else:
+            output = default
+
+        return self._clean_output(index, output, config) or default
+
+    def set_output(self, index, input, config):
```
Comment (qiluo-msft) [inline device/celestica/x86_64-cel_e1031-r0/sonic_platform/common.py:250]: >set_output [](http://example.com/codeflow?start=8&length=10)

Seems this function is not used. Please remove.
@mudsut4ke Could you help review?

**Pull request merged**